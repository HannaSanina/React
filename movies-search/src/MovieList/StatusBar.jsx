import React from 'react';
import styles from './movieList.css';
import { Button } from 'reactstrap';

const filters = ['release_date', 'vote_count'];

export default class StatusBar extends React.Component {
    sortChange = this.sortChange.bind(this);

    state = { isToggleOn: true };

    sortChange() {
      this.setState(prevState => ({ isToggleOn: !prevState.isToggleOn }));
      this.props.onSortChange(this.state.isToggleOn ? filters[0] : filters[1]);
    }

    render() {
      return (
        <div className={styles.resultStats}>
          <span>
            {this.props.result}
            {' '}
Movies found
          </span>
          <span className={styles.sortOption}>
            {' '}
Sort by
            <Button color="primary" onClick={e => this.sortChange(e)}>
              <span className="buttonLabel">
                {' '}
                {this.state.isToggleOn ? 'Release date' : 'Rating'}
              </span>
            </Button>
          </span>
        </div>
      );
    }
}
