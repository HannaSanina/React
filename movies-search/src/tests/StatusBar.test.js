import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StatusBar from '../MovieList/StatusBar';
configure({ adapter: new Adapter() });


describe('<StatusBar/>', () => {
    it('should be rendered correctly', () => {
        const onChange = jest.fn();
        const component = renderer.create(<StatusBar onSortChange={onChange} result={[movieDetails]}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('check button click', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<StatusBar onSortChange={onChange} result={[movieDetails]}/>);
        expect(wrapper.find('button').length).toEqual(1);
        wrapper.find('button').simulate('click');
        expect(onChange).toBeCalled();
    })
});

const movieDetails = new Array() [
    {
        "id": 0,
        "picture": "https://imgc.allpostersimages.com/img/posters/titanic_u-L-F4S6CQ0.jpg?src=gp&w=300&h=375",
        "title": "CANOPOLY",
        "about": "Fugiat velit commodo sint sit occaecat consequat magna nulla cupidatat incididunt magna. Veniam aliquip fugiat minim nulla incididunt aliqua dolore proident pariatur excepteur. Sunt pariatur enim minim magna. Do id dolor sint labore sint do eiusmod est et do ullamco aliquip.",
        "releaseDate": "January 6, 2014"
    }]