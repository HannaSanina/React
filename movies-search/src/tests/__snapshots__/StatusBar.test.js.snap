// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar/> should be rendered correctly 1`] = `
<div
  className="resultStats"
>
  <span>
     Movies found
  </span>
  <span
    className="sortOption"
  >
     Sort by
    <button
      onClick={[Function]}
    >
      <span
        className="buttonLabel"
      >
         
        Release date
      </span>
    </button>
  </span>
</div>
`;
